pep-0597.rst

```diff
@@ -33,135 +34,140 @@ Using the default encoding is a common mistake
 Developers using macOS or Linux may forget that the default encoding
 is not always UTF-8.
 
-For example, ``long_description = open("README.md").read()`` in
-``setup.py`` is a common mistake. Many Windows users can not install
-the package if there is at least one non-ASCII character (e.g. emoji)
-in the ``README.md`` file which is encoded in UTF-8.
+For example, using ``long_description = open("README.md").read()`` in
+``setup.py`` is a common mistake. Many Windows users cannot install
+such packages if there is at least one non-ASCII character (e.g. emoji)
+in their UTF-8-encoded ``README.md`` file.
 
-For example, 489 packages of the 4000 most downloaded packages from
-PyPI used non-ASCII characters in README. And 82 packages of them
-can not be installed from source package when locale encoding is
-ASCII. [1]_ They used the default encoding to read README or TOML
-file.
+Of the 4000 most downloaded packages from PyPI, 489 use non-ASCII
+characters in their README, and 82 of these do not explicitly specify
+the encoding used to open it. [1]_ Therefore, they cannot be installed
+from source in non-UTF-8 locales.
```

Comment:
Here you previously mentioned loading "TOML", but I was very confused what you were referring to as it didn't connect to the previous sentance and the only commonly-used TOML file I'm aware of in packaging is `pyproject.toml`, which AFAIK is not loaded programmatically in setup.py. Is there something I'm missing here?

Reply:
The bug in PEP 517 backend. And it is still alive. See here. https://github.com/takluyver/intreehooks/blob/4d6bdc6445cc84fb9774d0084639e09b308d2615/intreehooks.py#L31-L34

One year ago, `pip install poetry --no-binary :all:` failed in C locale because of this bug. But this backend is not used for now so it works for now.

Reply:
Ah, I see, thanks. Does the 82 uninstallable packages include those affected by that upstream bug, or is that just counting those with issues in their own `setup.py` (e.g. loading long_description from README)? Is it worth describing in enough detail that readers can understand what you're referring to, or do you have somewhere you can link it?

Reply:
They are included.

Reply:
Gotcha, so do we want to mention it with least link it? I couldn't find a canonical link to the upstream issue you mentioned, just downstream reports, so help there would be appreciated. What about revising the last three lines to read like this (I can't make it as a suggestion here, but if you agree I can do it separately):

```rst
characters in their README, and 82 fail to install on non-UTF-8 locales
caused by not specifying a file encoding [1]_, either loading the README
or loading the ``pyproject.toml`` in the PEP 517 build backend [LINK]_.
```

Reply:
Do we need mentioning all details here?

Intreehooks caused error: poetry
Packages using poetry during install: xdg, flake8-broken-line

pbr.packaging uses the default encoding. It caused install error of: stestr, 
https://opendev.org/openstack/pbr/src/branch/master/pbr/packaging.py#L826-L830

All other errors are caused by reading README or CHANGELOG in setup.py

Concrete logs is here: https://gist.github.com/methane/697ea72ce1cb62f1dbf8da7aaca5a6e9

Reply:
```
characters in their README, and 82 fail to install from source on
non-UTF-8 locales due to not specifying an encoding for a non-ASCII
file. [1]_
```

Totally up to you how detailed you want it; I just included that based on what you said previously, sorry. Do you prefer the above? Or something else?

Reply:
Applied the above as a suggestion, thanks!